dap/filter.go

```diff
@@ -0,0 +1,23 @@
+package dap
+
+import "github.com/TBD54566975/dap-go/maddr"
+
+// MoneyAddressFilter accepts a maddr.MoneyAddress, tests if it matches the filter condition, and returns either a more
+// specific money address type, or nil
+type MoneyAddressFilter[T any] func(maddr maddr.MoneyAddress) (*T, error)
+
+// Filter applies the MoneyAddressFilter to each element in maddrs, and returns a slice of the MoneyAddresses that
+// matched
+func Filter[T any](maddrs []maddr.MoneyAddress, filter MoneyAddressFilter[T]) ([]T, error) {
```

Comment:
putting `Filter` in the `maddr` package would likely provide more context as to what this is a filter for without having to look at the argument types. `maddr.Filter` vs. `dap.Filter`. the latter makes me think i'm filtering daps